inferlo/forney/optimization/nfg_map_lp.py

```diff
@@ -0,0 +1,98 @@
+from collections import namedtuple
+from itertools import product
+import cvxpy as cp
+import numpy as np
+
+
+from inferlo.forney.nfg_model import NormalFactorGraphModel
+
+map_lp_result = namedtuple('map_lp_result', ['upper_bound',
+                                             'lower_bound',
+                                             'factor_beliefs',
+                                             'variable_beliefs',
+                                             'normalization_duals',
+                                             'marginalization_duals'
+                                             ])
+
+
+def map_lp(model: NormalFactorGraphModel) -> map_lp_result:
+    """MAP linear programming relaxation for
```

Comment:
Please explain in more detail what this function does.

If this is algorithm from a paper, please add reference to paper. 

Document arguments and what it returns.

Please expand abbreviations (MAP). Not erveryone reading it will know what it means.